src/fsort.c

```diff
@@ -252,11 +252,13 @@ SEXP fsort(SEXP x, SEXP verboseArg) {
 
     if (verbose) {
       Rprintf(_("Top 20 MSB counts: ")); for(int i=0; i<MIN(MSBsize,20); i++) Rprintf(_("%"PRId64" "), (int64_t)msbCounts[order[i]]); Rprintf(_("\n"));
-      Rprintf(_("Reduced MSBsize from %zu to "), MSBsize);
     }
-    while (MSBsize>0 && msbCounts[order[MSBsize-1]] < 2) MSBsize--;
-    if (verbose) {
-      Rprintf(_("%zu by excluding 0 and 1 counts\n"), MSBsize);
+    {
```

Comment:
what does a "naked" braced statement like this do in C? Is it just restricting the scope of `oldMSBsize`? Does the variable then get freed right afterward?

Reply:
Exactly, this is to limit the scope of `oldMSBsize`. In particular, changes to `MSBsize` still propagate because it's declared in a different scope. A reasonable C compiler would probably adjust the stack pointer after the closing brace to make `oldMSBsize` disappear.